pkg/sysconfwatch/sysconfwatch_linux.go

```diff
@@ -92,6 +104,17 @@ func (c *SysConfWatchController) Run(ctx context.Context, ready chan<- struct{},
 	// Take a snapshot of the system and monototic clocks as a base reference
 	stimeRef, mtimeRef := getSysMonTimes()
 
+	for i := range c.iptClients {
+		iptClient := c.iptClients[i]
+		go iptClient.Monitor(
+			utiliptables.Chain("MICROSHIFT-SYSCONF-CANARY"),
+			[]utiliptables.Table{utiliptables.TableMangle, utiliptables.TableNAT, utiliptables.TableFilter},
+			func() { klog.Warningf("iptables flush is detected, restarting MicroShift"); os.Exit(0) },
+			1*time.Minute,
```

Comment:
Should we be using the same sysConfigCheckInterval here?

Reply:
My concern was it might be too frequent to query the iptable rules, but having a short interval shall work I think.
Updated to use the sysConfigCheckInterval.